src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -3,19 +3,202 @@
+        EXPECT_EQ(ringBuffer.getLength(), length);
+    }
+
+    // Test state after initialization and reset.
+    TEST_F(RingBufferTest, Reset) {
+        RingBuffer<float> ringBuffer(512);
+
+        // Check initial state.
+        EXPECT_EQ(ringBuffer.getReadIdx(), 0);
+        EXPECT_EQ(ringBuffer.getWriteIdx(), 0);
+        for (int i = 0; i < ringBuffer.getLength(); i++) {
+            EXPECT_EQ(ringBuffer.get(i), 0.0f);
+        }
+
+        // Fill ring buffer, mess with indices.
+        const float fill = 123.456f;
+        for (int i = 0; i < ringBuffer.getLength(); i++) {
+            ringBuffer.putPostInc(fill);
+            EXPECT_EQ(ringBuffer.get(i), fill);
+        }
+
+        ringBuffer.setWriteIdx(17);
+        ringBuffer.setReadIdx(42);
+
+        // Check state after reset.
+        ringBuffer.reset();
+        EXPECT_EQ(ringBuffer.getReadIdx(), 0);
+        EXPECT_EQ(ringBuffer.getWriteIdx(), 0);
+        for (int i = 0; i < ringBuffer.getLength(); i++) {
+            EXPECT_EQ(ringBuffer.get(i), 0.0f);
+        }
+    }
+
+    // Test inputs to setWriteIdx/setReadIdx outside of bounds [0, length).
+    TEST_F(RingBufferTest, Bounds) {
+        RingBuffer<float> ringBuffer(5);
+
+        ringBuffer.setWriteIdx(5);
+        EXPECT_EQ(ringBuffer.getWriteIdx(), 0);
+        ringBuffer.setWriteIdx(17);
+        EXPECT_EQ(ringBuffer.getWriteIdx(), 2);
+        ringBuffer.setWriteIdx(-2);
+        EXPECT_EQ(ringBuffer.getWriteIdx(), 3);
+
+        ringBuffer.setReadIdx(5);
+        EXPECT_EQ(ringBuffer.getReadIdx(), 0);
+        ringBuffer.setReadIdx(17);
+        EXPECT_EQ(ringBuffer.getReadIdx(), 2);
+        ringBuffer.setReadIdx(-2);
+        EXPECT_EQ(ringBuffer.getReadIdx(), 3);
+    }
+
+    // Test as a delay using a random signal.
+    TEST_F(RingBufferTest, Signal) {
+        const int signalLength = 1024;
+        int signal[signalLength];
+        RingBuffer<int> ringBuffer(256);
+
+        // Generate random signal.
+        srand(time(NULL));
+        for (int i = 0; i < signalLength; i++) {
+            signal[i] = rand();
+        }
+
+        // Feed signal to ring buffer.
+        // Wait for `delay` to start taking values out,
+        // and then ensure the signal is delayed as expected.
+        int delay = 100;
+        for (int i = 0; i < ringBuffer.getLength() * 4; i++) {
+            ringBuffer.putPostInc(signal[i]);
+            if (i >= delay) {
+                EXPECT_EQ(ringBuffer.getPostInc(), signal[i - delay]);
+            }
+        }
+    }
+
+    // Test fractional offset.
+    TEST_F(RingBufferTest, Fraction) {
+        RingBuffer<float> ringBuffer(3);
+        ringBuffer.putPostInc(17);
+        ringBuffer.putPostInc(40);
+        ringBuffer.putPostInc(-5);
+        EXPECT_EQ(ringBuffer.get(0.5f), (17 + 40)/2.0);
+        EXPECT_EQ(ringBuffer.get(1.5f), (40 - 5)/2.0);
+        EXPECT_EQ(ringBuffer.get(-0.5f), (17 - 5)/2.0);
+    }
+
+    class VibratoTest: public testing::Test {
```

Comment:
All tests under this class do not seem to be implemented, and (naturally) fails when the test suite is run

Reply:
Thanks for the heads up! (John wrote the vibrato tests [a while ago](https://github.com/ijc8/2022-MUSI6106/commit/6fad79118e35147a7b2c5cc747be19f64593d1b3#diff-d8415f8468352cb154de65ce8b058377c7ab80e1bab52eb9c7c36f07a68db64f), but we had a merge that went awry. :upside_down_face:)